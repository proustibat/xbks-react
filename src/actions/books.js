// SET_BOOKS
export const setBooks = books => ( {
    type: 'SET_BOOKS',
    books
} );
