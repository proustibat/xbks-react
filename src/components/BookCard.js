import React from 'react';
import numeral from 'numeral';

export default class BookCard extends React.Component {
    constructor( props ) {
        super( props );

        this.state = {
            isbn: props.book ? props.book.isbn : '',
            title: props.book ? props.book.title : '',
            price: props.book ? ( props.book.price ).toString() : '',
            cover: props.book ? props.book.cover : '',
            synopsis: props.book ? props.book.synopsis : []
        }
    }

    render() {
        return (
            <div className="bookcard">
                <h1 className="bookcard__title">{ this.state.title }</h1>

                <div className="bookcard__row">
                    <div className="bookcard__cell-cover">
                        <img className="bookcard__cover" src={ this.state.cover } alt={ `cover for ${ this.state.title }` } />
                        <p className="bookcard__price">
                            <span>{ numeral( this.state.price ).format( '0.00$' ) }</span>
                        </p>
                    </div>
                    <div className="bookcard__cell-details">
                        <div className="bookcard__details">
                            <div className="bookcard__description">{ this.state.synopsis.join( ' ' ) }</div>
                        </div>
                    </div>
                </div>

            </div>
        );
    }
}

