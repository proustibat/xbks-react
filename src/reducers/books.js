const booksReducerDefaultState = [];
export default ( state = booksReducerDefaultState, action ) => {
    switch ( action.type ) {
        case 'SET_BOOKS':
            return action.books;
        default:
            return state;
    }
};
