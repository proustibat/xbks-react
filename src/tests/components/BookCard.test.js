import React from 'react';
import { shallow } from 'enzyme';
import BookCard from '../../components/BookCard';
import books from "../fixtures/books";

test( 'Should render BookCard component correctly with no data', () => {
    const wrapper = shallow( <BookCard /> );
    expect( wrapper ).toMatchSnapshot();
} );

test( 'Should render BookCard component correctly with data', () => {
    const wrapper = shallow( <BookCard
        book={ books[ 0 ] }
    /> );

    expect( wrapper ).toMatchSnapshot();
} );
