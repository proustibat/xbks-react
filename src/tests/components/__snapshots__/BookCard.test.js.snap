// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render BookCard component correctly with data 1`] = `
<div
  className="bookcard"
>
  <h1
    className="bookcard__title"
  >
    Henri Potier à l'école des sorciers
  </h1>
  <div
    className="bookcard__row"
  >
    <div
      className="bookcard__cell-cover"
    >
      <img
        alt="cover for Henri Potier à l'école des sorciers"
        className="bookcard__cover"
        src="http://henri-potier.xebia.fr/hp0.jpg"
      />
      <p
        className="bookcard__price"
      >
        <span>
          35.00$
        </span>
      </p>
    </div>
    <div
      className="bookcard__cell-details"
    >
      <div
        className="bookcard__details"
      >
        <div
          className="bookcard__description"
        >
          Après la mort de ses parents (Lily et James Potier), Henri est recueilli par sa tante Pétunia (la sœur de Lily) et son oncle Vernon à l'âge d'un an. Ces derniers, animés depuis toujours d'une haine féroce envers les parents du garçon qu'ils qualifient de gens « bizarres », voire de « monstres », traitent froidement leur neveu et demeurent indifférents aux humiliations que leur fils Dudley lui fait subir. Henri ignore tout de l'histoire de ses parents, si ce n'est qu'ils ont été tués dans un accident de voiture Le jour des 11 ans de Henri, un demi-géant du nom de Rubeus Hagrid vient le chercher pour l’emmener à Poudlard, une école de sorcellerie, où il est inscrit depuis sa naissance et attendu pour la prochaine rentrée. Hagrid lui révèle alors qu’il a toujours été un sorcier, tout comme l'étaient ses parents, tués en réalité par le plus puissant mage noir du monde de la sorcellerie, Voldemort (surnommé « Celui-Dont-On-Ne-Doit-Pas-Prononcer-Le-Nom »), après qu'ils ont refusé de se joindre à lui. Ce serait Henri lui-même, alors qu'il n'était encore qu'un bébé, qui aurait fait ricocher le sortilège que Voldemort lui destinait, neutralisant ses pouvoirs et le réduisant à l'état de créature quasi-insignifiante. Le fait d'avoir vécu son enfance chez son oncle et sa tante dépourvus de pouvoirs magiques lui a donc permis de grandir à l'abri de la notoriété qu'il a dans le monde des sorciers. Henri entre donc à l’école de Poudlard, dirigée par le professeur Albus Dumbledore. Il est envoyé dans la maison Gryffondor par le « choixpeau ». Il y fait la connaissance de Ron Weasley et Hermione Granger, qui deviendront ses complices. Par ailleurs, Henri intègre rapidement l'équipe de Quidditch de sa maison, un sport collectif très populaire chez les sorciers se pratiquant sur des balais volants. Henri connaît probablement la plus heureuse année de sa vie, mais également la plus périlleuse, car Voldemort n'a pas totalement disparu et semble bien décidé à reprendre forme humaine.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Should render BookCard component correctly with no data 1`] = `
<div
  className="bookcard"
>
  <h1
    className="bookcard__title"
  />
  <div
    className="bookcard__row"
  >
    <div
      className="bookcard__cell-cover"
    >
      <img
        alt="cover for "
        className="bookcard__cover"
        src=""
      />
      <p
        className="bookcard__price"
      >
        <span>
          0.00$
        </span>
      </p>
    </div>
    <div
      className="bookcard__cell-details"
    >
      <div
        className="bookcard__details"
      >
        <div
          className="bookcard__description"
        />
      </div>
    </div>
  </div>
</div>
`;
